accepted/2024/improve-support-for-packages-signed-with-trusted-signing.md

```diff
@@ -0,0 +1,155 @@
+step 1 the certificate would no longer be shown and the user would only see their EKU's. The Subject and Issuer will be extracted
+from the certificate the first time and added to the EKU data. Future uploads should also check the Subject and Issuer and update
+the data if they have changed.
+
+#### How does this help the user?
+
+When only step 1 is implemented it will already be a huge help for users that use Trusted Signing but it will look strange inside
+the NuGet Gallery because no certificates are shown anymore and they cannot unlink their EKU. Adding this separate table with EKU's
+linked to the user their account would allow them to remove this EKU. I personally don't think users would really need this
+functionality right away so that is why I think this could be added as a second step.
+
+### Step 3: Allow a user to submit a Public Trust identity EKU in the NuGet Gallery
+
+#### How would this be implemented?
+
+When a `.cer` file is uploaded only the thumbprint is stored because the certificate is not parsed by the NuGet Gallery due to
+security reasons. This means we cannot add the Public trust identity EKU record when a certificate is uploaded. For a user that
+has access to the Azure Portal it would be easier to copy the EKU from the interface:
+
+![Azure Portal](images/trusted-signing-azure.png)
+
+And this can also be done before a NuGet package has been signed. This copied value should then be entered into a form that adds
+this EKU to the user. As mentioned earlier the Subject and Issuer will be set once the first package that has this EKU is uploaded.
+
+#### How does this help the user?
+
+Adding this in the interface would make it possible for a user to register an EKU in their account. This would be easier than
+extracting the certificate from a signed NuGet package. The process to get that `.cer` file from a signed package is not that
+simple. This would be the final step that would give a user full control over their EKU's.
+
+### Technical explanation
+
+<!-- Explain the proposal in sufficient detail with implementation details, interaction models, and clarification of corner cases. -->
+
+## Drawbacks
+
+<!-- Why should we not do this? -->
+
+## Rationale and alternatives
+
+<!-- Why is this the best design compared to other designs? -->
+<!-- What other designs have been considered and why weren't they chosen? -->
+<!-- What is the impact of not doing this? -->
+
+## Prior Art
+
+<!-- What prior art, both good and bad are related to this proposal? -->
+<!-- Do other features exist in other ecosystems and what experience have their community had? -->
+<!-- What lessons from other communities can we learn from? -->
+<!-- Are there any resources that are relevant to this proposal? -->
+
+## Unresolved Questions
+
+<!-- What parts of the proposal do you expect to resolve before this gets accepted? -->
```

Comment:
How do we (NuGet.org) get a Trusted Signing certificate for testing this feature?

Reply:
I can help with manual testing since I am personally onboarded. I am not sure if we need regular E2E testing (needing a new Trusted Signing signature per test run). Are you referring to manual or automated testing?

Reply:
> I can help with manual testing since I am personally onboarded.

Suppose you go on leave for a while.

> Are you referring to manual or automated testing?

Neither specifically at this time, though automated testing is preferrable.  I'm thinking about E2E testability during development, release, and post-release.